### PR TITLE
Add seed for train-test split.

### PR DESCRIPTION
**Pull request by ita9naiwa**
#409

### CODE REVIEW

File: implicit/evaluation.pyx [hunk 1]
```diff
@@ -14,7 +14,7 @@ from libc.math cimport fmin
 from libcpp.unordered_set cimport unordered_set
 
 
-def train_test_split(ratings, train_percentage=0.8):
+def train_test_split(ratings, train_percentage=0.8, seed=None):
```
Comment (benfred) [inline implicit/evaluation.pyx:17]: Our other API's take a 'random_state' parameter - that is either a np.random.RandomState or an integer (using this function to check/validate https://github.com/benfred/implicit/blob/master/implicit/utils.py#L34 to validate etc).  We should be consistent with other functions here

**Review by benfred — CHANGES_REQUESTED** [review]

File: implicit/evaluation.pyx (hunk 1 above)
Comment (ita9naiwa) [reply implicit/evaluation.pyx]: Thanks for that I've missed. I changed the code to use `check_random_state` defined in `implicit/util.py`.

**Comment by ita9naiwa** [comment]
Seems that the error is not related to the code I wrote...

File: implicit/evaluation.pyx [hunk 2]
```diff
@@ -1,20 +1,19 @@
...
+import numpy as np
 from cython.operator import dereference
 from cython.parallel import parallel, prange
-from libc.stdlib cimport malloc, free
-from libc.string cimport memset
-from libc.math cimport fmin
+from scipy.sparse import coo_matrix, csr_matrix
+from tqdm.auto import tqdm
+
+from libc.stdlibcimportmalloc import free
```
Comment (benfred) [inline implicit/evaluation.pyx:11]: The build is failing because of some of these import changes - I think this will fix?

```suggestion
from libc.stdlib cimport malloc, free
from libcpp.unordered_set cimport unordered_set 
```

File: implicit/evaluation.pyx [hunk 3]
```diff
@@ -1,20 +1,20 @@
...
+import numpy as np
 from cython.operator import dereference
 from cython.parallel import parallel, prange
+from scipy.sparse import coo_matrix, csr_matrix
+from tqdm.auto import tqdm
+
 from libc.stdlib cimport malloc, free
-from libc.string cimport memset
-from libc.math cimport fmin
+from libcpp.unordered_set cimport unordered_set
```
Comment (benfred) [inline implicit/evaluation.pyx:12]: Err- missed the other two imports 
```suggestion
from libc.string cimport memset
from libc.math cimport fmin
from libcpp.unordered_set cimport unordered_set 
```

File: implicit/evaluation.pyx [hunk 4]
```diff
@@ -1,20 +1,22 @@
...
+import numpy as np
 from cython.operator import dereference
 from cython.parallel import parallel, prange
+from scipy.sparse import coo_matrix, csr_matrix
+from tqdm.auto import tqdm
+
 from libc.stdlib cimport malloc, free
 from libc.string cimport memset
 from libc.math cimport fmin
+from libcpp.unordered_set cimport unordered_set
```
Comment (benfred) [inline implicit/evaluation.pyx:14]:
```suggestion
from libcpp.unordered_set cimport unordered_set
```

**Review by benfred — APPROVED** [review]

**Comment by kmcmearty** [comment]
I know I can pull this code from the repo but since I am using Conda as my manager, do I have to wait till the next Conda package update to get this functionality?
